app/models/container.rb

```diff
@@ -30,8 +30,8 @@ def event_where_clause(assoc = :ems_events)
     case assoc.to_sym
     when :ems_events
       # TODO: improve relationship using the id
-      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ?", container_project.name,
-       ext_management_system.id]
+      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ? AND container_name = ?",
+       container_project.name, ext_management_system.id, name]
```

Comment:
@dkorn @moolitayer we'll have to move and use the ids but not for this release. It's in the backlog.